plugins/zstack.py

```diff
@@ -0,0 +1,410 @@
+        logging.debug("Preparing Z Stack acquisition. Moving focus to start position")
+        self.old_pos = self.focus.position.value
+        self.focus.moveAbs({'z': self.zstart.value}).result()
+        speed = self.focus.speed.value['z']
+        return driver.estimateMoveDuration(abs(self.zstep.value), speed, 0.01)
+
+    def stepAcquisition(self, i, images):
+        self.focus.moveRel({'z': self.zstep.value}).result()
+        
+    def completeAcquisition(self, completed):
+        # Mvoe back to start
+        if completed:
+            logging.info("Z Stack acquisiition complete.")
+        logging.debug("Returning focus to start position %s", self.old_pos)
+        self.focus.moveAbs(self.old_pos).result()
+        
+    def postProcessing(self, images):
+        cubes = [self.constructCube(im) for im in images]
+        return cubes
+
+    def acquire(self, dlg):
+        main_data = self.main_app.main_data
+        str_ctrl = main_data.tab.value.streambar_controller
+        stream_paused = str_ctrl.pauseStreams()
+        dlg.pauseSettings()
+
+        nb = self.numberofAcquisitions.value
+        ss, last_ss = self._get_acq_streams()
+
+        sacqt = acq.estimateTime(ss)
+        
+        completed = False
+
+        try:
+            step_time = self.initAcquisition()
+            logging.debug("Acquisition streams: %s", ss)
+
+            # TODO: if drift correction, use it over all the time
+            f = model.ProgressiveFuture()
+            f.task_canceller = lambda l: True  # To allow cancelling while it's running
+            f.set_running_or_notify_cancel()  # Indicate the work is starting now
+            dlg.showProgress(f)
+
+            images = None
+        
+            for i in range(nb):
+                left = nb - i
+                dur = sacqt * left + step_time * (left - 1)
+
+                logging.debug("Acquisition %d of %d", i, nb)
+
+                if left == 1 and last_ss:
+                    ss += last_ss
+                    dur += acq.estimateTime(ss) - sacqt
+
+                startt = time.time()
+                f.set_progress(end=startt + dur)
+                das, e = acq.acquire(ss).result()
+                if images is None:
+                    # Copy metadata from the first acquisition
+                    self._metadata = copy.copy(das[0].metadata)
+                    images = [[] for i in range(len(das))]
+
+                for im, da in zip(images, das):
+                    im.append(da)
+
+                if f.cancelled():
+                    raise CancelledError()
+
+                # Execute an action to prepare the next acquisition for the ith acquisition
+                self.stepAcquisition(i, images)
+
+            f.set_result(None)  # Indicate it's over
+            
+            # Construct a cube from each stream's image.
+            images = self.postProcessing(images)
+
+            # Export image
+            logging.debug("Exporting Z Stack to HDF5")
```

Comment:
The export format depends on what the user picked. Either drop the "HDF5" completely, or use `exporter.format`.